### PR TITLE
fix: always append init segment after trackinfo change

### PR DESCRIPTION
**Pull request by brandonocasey**
This pull request does three things:

1. Always append an init segment after a `trackinfo` update. We do this because if the `trackinfo` causes a `changeType` we **have** to append an init segment or chrome will media error decode.
2. Check for individual segment trackinfo before appending and allow appending after trackinfo
3. Check for an abort after a trackinfo event is triggered, in the case where the type we are changing to is not supported.

### CODE REVIEW

File: src/segment-loader.js [hunk 1]
```diff
@@ -1895,7 +1914,7 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     this.sourceUpdater_.appendBuffer({segmentInfo, type, bytes}, (error) => {
       if (error) {
-        this.error(`appenderror for ${type} append with ${bytes.length} bytes`);
+        this.error(`append of ${bytes.length} failed for ${type} data of segment #${segmentInfo.mediaIndex} for playlist ${segmentInfo.playlist.id}`);
```
Comment (brandonocasey) [inline src/segment-loader.js:1917]: more descriptive error message about the failure

File: src/segment-loader.js [hunk 2]
```diff
@@ -1895,7 +1914,7 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     this.sourceUpdater_.appendBuffer({segmentInfo, type, bytes}, (error) => {
       if (error) {
-        this.error(`appenderror for ${type} append with ${bytes.length} bytes`);
+        this.error(`${type} append of ${bytes.length}b failed for segment #${segmentInfo.mediaIndex} in playlist ${segmentInfo.playlist.id}`);
```
Comment (gkatsev) [inline src/segment-loader.js:1917]: 👍

Comment (gesinger) [reply src/segment-loader.js:1917]: Double 👍

File: src/segment-loader.js [hunk 3]
```diff
@@ -1453,11 +1453,30 @@ export default class SegmentLoader extends videojs.EventTarget {
...
+        audio: true,
+        video: true
+      };
+
       this.startingMedia_ = trackInfo;
       this.logger_('trackinfo update', trackInfo);
       this.trigger('trackinfo');
     }
 
+    // trackinfo may cause an abort
```
Comment (gesinger) [inline src/segment-loader.js:1466]: Minor: but might be worth mentioning one of the example cases

File: src/segment-loader.js [hunk 4]
```diff
@@ -1678,7 +1697,7 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     const segmentInfo = this.pendingSegment_;
 
-    if (!segmentInfo || !this.startingMedia_) {
+    if (!segmentInfo || !segmentInfo.trackInfo) {
       // no segment to append any data for
```
Comment (gesinger) [inline src/segment-loader.js:1701]: Might be worth an update to this comment

**Pull request merged**